### DISCUSSION TITLE
Cannot import dynamically named exports in 13.2.4

### DISCUSSION THREAD

**Discussion by conorhub** [vercel/next.js#47316] [Help]
### Summary

From v13.2.4 I'm getting the error:

`Error: Cannot access .then on the server. You cannot dot into a client module from a server component. You can only pass the imported name through.`

When attempting to import a dynamically named export via code that was working in 13.1+

```
const MapPanel = dynamic(
	() => import('@/components/Events/MapPanel')
	.then((mod) => mod.MapPanel), { 
		ssr: false,
		loading: () => 'Loading...'
	}
);
```

The beta docs still [seem to suggest ](https://beta.nextjs.org/docs/optimizing/lazy-loading#example-importing-named-exports) using the .then method, so I'm unsure as to what I can do to adjust the code to work in 13.2.4.

### Additional information

The above occurs on a /app server-side `page.jsx`.

`MapPanel` is a client-side component that contains code that imports and renders `react-leaflet` components. `React-leaflet` requires the `ssr: false` flag for compatibility as it will fail otherwise (aspects of it will attempt to render on the server if not set).

**Comment by conorhub** [comment]
I was able to modify the MapPanel component to work without having to use .then() in page.jsx. While the topic can be closed, I'm still curios to know why .then() threw this error and failed to build in 13.2.4 and not prior versions. I didn't see it mentioned in changes.

**Comment by Pbmangan** [comment]
This error is also thrown if you have Head.ts/js files with 13.2. Was able to resolve when head.js were removed and metadata was moved to the new metadata structure.

**Reply by conorhub** [reply]
I can't verify this as I no longer use the above code, but this could be what caused it. I was using head.js files (with the intention of migrating from them, as the deprecation warning was quite apparent)

**Reply by eric-burel** [reply]
This happens to me when using Lodash cloneDeep, quite surprising. This error appears on a specific page, then the app stops working for all pages. I am not using a dynamic import. Will try head thing and update this message if I have anything new.

Update: temporary disabling head.js files make this error disappear. I have another error message but it may be unrelated.
Update2: I have a page that tries to pass an object down to a Client component and that contains functions, which annoys Next. Not sure of the best approach, I've temporarily disabled it.
Update3: then I can reenable the "head" files

=> I suspect this error message is hiding another error related to trying to pass down non-serializable data to a client component. You should have a cleaner message, that happens when the page fails. But here, the "head" fails first and doesn't have the right error message.

I suggest turning that into an issue instead of a discussion, if a minimal repro can be created.

Update4: I could not craft a minimal repro yet :( adding "head" doesn't trigger this weird error message.

**Comment by nmitic** [comment]
This happens to me as well.

```
"next": "13.3.2",
```

I am having 
1) LoadingSpiner - with client side declaration with no props
2) Loading.tsx - where the above component is being dynamically imported

Loading.tsx looks like this:

```
import dynamic from "next/dynamic";

const LoadingIndicator = dynamic<{}>(
  () =>
    import("../../components/loadingIndicator/LoadingIndicator").then(
      (module) => module.LoadingIndicator
    ),
  { ssr: false }
);

const Loading = () => {
  return (
    <div>
      <LoadingIndicator />
    </div>
  );
};

export default Loading;
``` 

Error:

```Error: Cannot access .then on the server. You cannot dot into a client module from a server component. You can only pass the imported name through.```

Not sure how to solve this atm :D

**Reply by nmitic** [reply]
Moving from named to default export of dynamically imported cpm the error has been resolved.

**Reply by flbn** [reply]
can confirm, renaming to default fixed it for me

**Comment by fredowisk** [comment]
If you don't want to change every component to default export, you can do something like this:

```Javascript
import dynamic from "next/dynamic";

const ComponentName = dynamic(async () => {
  const { ComponentName: Component } = await import("@/components/ComponentName");
  return { default: Component };
}
```
